### ISSUE TITLE
training for my own dataset

### ISSUE THREAD

**Issue by NerminSalem** [MathiasGruber/PConv-Keras#6]
thanks alot for sharing yr code.
how can i use it to train on my own dataset ?
yr help is really appropriated

**Comment by KhyatiGanatra** [comment]
> thanks alot for sharing yr code.
> how can i use it to train on my own dataset ? also i have a question regarding used loss function, from wt i understand of yr code is that u only used root mean sqaure loss, is that correct?
> yr help is really appropriated

I have the same question. How to train on my custom dataset. It is trying to load saved models, which does not exist already. Any help will be appreciated

**Comment by NerminSalem** [comment]
@MathiasGruber your help is really appreciated, how to run the code?

**Comment by MathiasGruber** [comment]
You should follow the notebook "Step4 - Imagenet Training.ipynb", and change the TRAIN_DIR, TEST_DIR and VAL_DIR as appropriate. Let me know if it doesn't work :)

**Comment by KhyatiGanatra** [comment]
> :)

Yes, thanks for the reply. I have changed the directories. But in later part where you are trying to load model, and trying to access logs, its empty. There is no model to load in Phase 1. Please correct if I am wrong

**Comment by NerminSalem** [comment]
> You should follow the notebook "Step4 - Imagenet Training.ipynb", and change the TRAIN_DIR, TEST_DIR and VAL_DIR as appropriate. Let me know if it doesn't work :)

thanks alot, i figured it out, however when to run step 4 it is trying to load Github-Public\PConv-Keras\data\logs\170_weights_2018-06-28-15-00-38.h5 which doesn't exist from where can i download them
yr help is really appreciated

**Comment by NerminSalem** [comment]
> You should follow the notebook "Step4 - Imagenet Training.ipynb", and change the TRAIN_DIR, TEST_DIR and VAL_DIR as appropriate. Let me know if it doesn't work :)

i m getting this error (Found 0 images belonging to 0 classes.). do i have to put the images in a folder and wt should i name it? also i followed the same paths to train, pexels and val
yr help is really needed thanks

**Comment by Zzzzy411** [comment]
> i m getting this error (Found 0 images belonging to 0 classes.). do i have to put the images in a folder and wt should i name it? also i followed the same paths to train, pexels and val
> yr help is really needed thanks

Have you solved this problem ？Thanks for your answer.

**Comment by NerminSalem** [comment]
Sadly still didn't find a solution

**Comment by xdujiang** [comment]
@NerminSalem @Zzzzy411 
There must be subfolders in your TRAIN_DIR, TEST_DIR, VAL_DIR, and you need to put image files in the subfolders...
https://github.com/keras-team/keras/issues/3946#issuecomment-400546787

**Comment by mojc** [comment]
The `model.load` is not implemented in the current version. The code cannot work since it tries to load weights from the logs and then calls a non implemented method load with the weight path. this seems weird. Was there an earlier version where this worked?

**Comment by JaimeCarracedo** [comment]
> > :)
> 
> Yes, thanks for the reply. I have changed the directories. But in later part where you are trying to load model, and trying to access logs, its empty. There is no model to load in Phase 1. Please correct if I am wrong

Hi, have you any solution for this problem? Thanks.

**Comment by ZDD2009** [comment]
> You should follow the notebook "Step4 - Imagenet Training.ipynb", and change the TRAIN_DIR, TEST_DIR and VAL_DIR as appropriate. Let me know if it doesn't work :)
> 
> i m getting this error (Found 0 images belonging to 0 classes.). do i have to put the images in a folder and wt should i name it? also i followed the same paths to train, pexels and val
> yr help is really needed thanks
Hi,Had the problem been solved?Thanks!

**Comment by ZDD2009** [comment]
i m getting this error (Found 0 images belonging to 0 classes.). do i have to put the images in a folder and wt should i name it? also i followed the same paths to train, pexels and val
yr help is really needed thanks

**Comment by MathiasGruber** [comment]
I've updated the repo with a CLI, so you should now be able to do: 
python main.py \
    --name MyDataset \
    --train TRAINING_PATH \
    --validation VALIDATION_PATH \
    --test TEST_PATH \
    --vgg_path './data/logs/pytorch_to_keras_vgg16.h5'

Note that each image folder needs one or more subfolders that contain the actual images, since I'm basing my image generator on the ImageDataGenerator.

**Comment by shabin75** [comment]
i have an issue
after first epoch
current_index = (self.batch_index * self.batch_size) % self.n
ZeroDivisionError: integer division or modulo by zero
pls help me

**Comment by Lvhhhh** [comment]
> i have an issue
> after first epoch
> current_index = (self.batch_index * self.batch_size) % self.n
> ZeroDivisionError: integer division or modulo by zero
> pls help me

i have the same problem , do you fix it?

**Comment by ttxxr** [comment]
> > i have an issue
> > after first epoch
> > current_index = (self.batch_index * self.batch_size) % self.n
> > ZeroDivisionError: integer division or modulo by zero
> > pls help me
> 
> i have the same problem , do you fix it?
I have the same problem，too. If you have any solution for it , could you tell me ,please ?

**Comment by Lvhhhh** [comment]
> > > i have an issue
> > > after first epoch
> > > current_index = (self.batch_index * self.batch_size) % self.n
> > > ZeroDivisionError: integer division or modulo by zero
> > > pls help me
> > 
> > 
> > i have the same problem , do you fix it?
> > I have the same problem，too. If you have any solution for it , could you tell me ,please ?

i fix it , delete the 'val' because there is no 'val' folder in your valiation sample

**Comment by Lvhhhh** [comment]
> i have an issue
> after first epoch
> current_index = (self.batch_index * self.batch_size) % self.n
> ZeroDivisionError: integer division or modulo by zero
> pls help me

delete 'val' in your valition sample

**Comment by TTXXR** [comment]
thanks a lot！

**Comment by elham1992** [comment]
thanks alot for sharing your code.
 i use the cli to train on my own dataset  .  
python main.py --name celeba --train ./data/train  --test ./data/test --vgg_path ./data/pytorch_to_keras_vgg16.h5

out put  on command prompt is :
![image](https://user-images.githubusercontent.com/36129307/56881486-ce843700-6a75-11e9-91a8-dc8f03e02197.png)
why no result  and training is 0%??
pleas help me and answer to my Question.

**Comment by SunnyPann** [comment]
> i have an issue
> after first epoch
> current_index = (self.batch_index * self.batch_size) % self.n
> ZeroDivisionError: integer division or modulo by zero
> pls help me

Are you training on your own training set?
Where can I download the training model?
thank you

**Comment by jiguanglu** [comment]
> > i m getting this error (Found 0 images belonging to 0 classes.). do i have to put the images in a folder and wt should i name it? also i followed the same paths to train, pexels and val
> > yr help is really needed thanks
> 
> Have you solved this problem ？Thanks for your answer.
Hi ,  
TRAIN_DIR = r'/home/user/my_test/PConv-Keras/data/train'
VAL_DIR = r'/home/user/my_test/PConv-Keras/data/val'
TEST_DIR = r'/home/user/my_test/PConv-Keras/data/test'
In each folder, you must have the sub folder and they must have the same name.  like this.

data/
...images/
........./images
...masks/
........./masks

**Comment by denizekiz** [comment]
Does cli automatically convert images to 512x512?
